estate/models/estate_property.py

```diff
@@ -81,3 +82,19 @@ def _onchange_garden(self):
         else:
             self.garden_area = 0
             self.garden_orientation = None
+
+    def action_state_sold(self):
+        for record in self:
+            if record.state != "cancelled":
+                record.state = "sold"
+            else:
+                UserError("A cancelled property cannot be set as sold")
```

Comment:
Could be shortened using mapped and batch assignment
```suggestion
        if "cancelled" in self.mapped('state'):
            UserError("A cancelled property cannot be set as sold")
        self.state = 'sold'
```